scikeras/wrappers.py

```diff
@@ -549,8 +565,14 @@ def _check_model_compatibility(self, y: np.ndarray) -> None:
                 for x in [self.loss, self.model_.loss]
             ):  # filter out loss list/dicts/etc.
                 if default_val is not None:
-                    default_val = loss_name(default_val)
-                given = loss_name(self.loss)
+                    try:
+                        default_val = loss_name(default_val)
+                    except ValueError:
+                        pass
```

Comment:
Todo: figure out what to do here, or even refactor this check like in #208

Reply:
I think `loss_name` returning `None` says "the provided loss has no name/is not recognized."

Reply:
I'll try to put that change in. I still think this check needs to be refactored

Reply:
This worked out. Only small doc and test changes needed.